### PR TITLE
allow download endpoint for any version matching current major

### LINKED ISSUE #1726: Download link in receipts does not trigger download
Getting multiple reports of this lately

### PR DESCRIPTION
**Pull request by btkostner**
Fixes #1726 

### Changes Summary

- Allows any os purchase with the same major version to download the current version from sent email receipt.

### CODE REVIEW

File: api/download.php [hunk 1]
```diff
@@ -22,40 +22,49 @@ function go_home() {
...
 
-$iso_version = false;
+$isoVersion = false;
 foreach ($products as $product) {
     if (substr($product, 0, 3) === 'ISO') {
-        $iso_version = substr($product, 4);
+        $isoVersion = substr($product, 4);
+        list($isoMajor) = explode('.', $isoVersion);
+
+        if ($isoMajor != null) {
```
Comment (lewisgoddard) [inline api/download.php]: What happens if this if fails?

File: api/download.php [hunk 2]
```diff
@@ -22,40 +22,49 @@ function go_home() {
...
+        return go_home();
     }
 }
 
-$iso_version = false;
+$isoVersion = false;
 foreach ($products as $product) {
     if (substr($product, 0, 3) === 'ISO') {
-        $iso_version = substr($product, 4);
+        $isoVersion = substr($product, 4);
```
Comment (lewisgoddard) [inline api/download.php]: Is this literally taking the first 4 characters? That won't work with double digits, why not just handle them with explode like you do later?

Comment (lewisgoddard) [inline api/download.php]: Ah, okay, that's set in the product definitions, right?

File: api/download.php (hunk 1 above)
Comment (btkostner) [inline api/download.php]: If we can't detect the major release version from the receipt, then we don't try to match it and only process the exact match. If for some reason it's like `ISO_release` then a purchase cookie for `release` gets set.

File: api/download.php (hunk 2 above)
Comment (btkostner) [inline api/download.php]: This is for the release payment. When you buy it we add meta data to the charge in stripe with a line like `ISO_0.4.1` to indicate that the 0.4.1 release was bought.

**Pull request merged**